component/src/main/java/io/siddhi/extension/store/mongodb/MongoSelectExpressionVisitor.java

```diff
@@ -0,0 +1,307 @@
+    public void beginVisitOrRightOperand() {
+    }
+
+    @Override
+    public void endVisitOrRightOperand() {
+    }
+
+    @Override
+    public void beginVisitAnd() {
+    }
+
+    @Override
+    public void endVisitAnd() {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$and:" + "[" + leftOperand + "," + rightOperand + "]}");
+    }
+
+    @Override
+    public void beginVisitAndLeftOperand() {
+    }
+
+    @Override
+    public void endVisitAndLeftOperand() {
+    }
+
+    @Override
+    public void beginVisitAndRightOperand() {
+    }
+
+    @Override
+    public void endVisitAndRightOperand() {
+    }
+
+    @Override
+    public void beginVisitIsNull(String streamId) {
+        this.isNullCheck = true;
+    }
+
+    @Override
+    public void endVisitIsNull(String streamId) {
+        this.isNullCheck = false;
+        String nullCheckAttribute = conditionOperands.pop();
+        conditionOperands.push("{ $cond: { if: { $eq: [" + nullCheckAttribute + ", null ] }, then: true, " +
+                "else: false } }");
+    }
+
+    @Override
+    public void beginVisitCompare(Compare.Operator operator) {
+    }
+
+    @Override
+    public void endVisitCompare(Compare.Operator operator) {
+        String compareFilter;
+        switch (operator) {
+            case EQUAL:
+                compareFilter = "$eq";
+                break;
+            case GREATER_THAN:
+                compareFilter = "$gt";
+                break;
+            case GREATER_THAN_EQUAL:
+                compareFilter = "$gte";
+                break;
+            case LESS_THAN:
+                compareFilter = "$lt";
+                break;
+            case LESS_THAN_EQUAL:
+                compareFilter = "$lte";
+                break;
+            case NOT_EQUAL:
+                compareFilter = "$ne";
+                break;
+            default:
+                throw new MongoTableException("MongoDB Event Table found unknown operator '" + operator + "' for " +
+                        "COMPARE operation.");
+        }
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{ $cond: { if: { " + compareFilter + ": [" + leftOperand + "," + rightOperand + " ] }, then: true," +
+                "else: false } }");
+    }
+
+    @Override
+    public void beginVisitCompareLeftOperand(Compare.Operator operator) {
+    }
+
+    @Override
+    public void endVisitCompareLeftOperand(Compare.Operator operator) {
+    }
+
+    @Override
+    public void beginVisitCompareRightOperand(Compare.Operator operator) {
+    }
+
+    @Override
+    public void endVisitCompareRightOperand(Compare.Operator operator) {
+    }
+
+    @Override
+    public void beginVisitNot() {
+        throw new MongoTableException("MongoDB Event Table does not support NOT function.");
```

Comment:
Not is supported in MongoDB, https://github.com/siddhi-io/siddhi-store-mongodb/blob/master/component/src/main/java/io/siddhi/extension/store/mongodb/util/MongoTableConstants.java#L53

Reply:
Fixed in
https://github.com/siddhi-io/siddhi-store-mongodb/pull/56/commits/19316e5642550ec5d4d154a11df69b7c14d8d726